### PR TITLE
cmake: Toolchain abstraction: Abstraction of compiler imacros option

### PR DESCRIPTION
**Pull request by daor-oti**
The imacros compiler option for inclusion of the autoconf.h header file
is compiler and thus toolchain specific, this PR allows for each 
toolchain to decide how the autoconf.h header file should be included.  

The intent here is to abstract Zephyr's dependence on toolchains,
thus allowing for easier porting to other, perhaps commercial, toolchains
and/or usecases.

No functional change expected.

Part of #16031

### CODE REVIEW

**Review by mbolivar — APPROVED** [review]
Looks reasonable to me.

**Review by mped-oticon — APPROVED** [review]
LGTM

**Review by SebastianBoe — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -214,12 +214,15 @@ endif()
...
+# @Intent: Set compiler specific macro inclusion of AUTOCONF_H
+toolchain_cc_imacros(${AUTOCONF_H})
+
 # @Intent: Set compiler specific flag for bare metal freestanding option
 toolchain_cc_freestanding()
 
 zephyr_compile_options(
   -g # TODO: build configuration enough?
-  -imacros ${AUTOCONF_H}
+
```
Comment (SebastianBoe) [inline CMakeLists.txt]: Is there a line of whitespace here?

Comment (daor-oti) [reply CMakeLists.txt]: @SebastianBoe Yes, sorry for that, now removed.

**Pull request merged**